controllers/azuremanagedmachinepool_reconciler.go

```diff
@@ -112,7 +114,14 @@ func (s *azureManagedMachinePoolService) Reconcile(ctx context.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to get agent pool parameters")
 	}
-	agentPoolName := agentPool.AzureName()
+	var agentPoolName string
+	if s.scope.IsPreviewEnabled() {
+		agentPoolTyped := agentPool.(*asocontainerservicev1preview.ManagedClustersAgentPool)
```

Comment:
nit: do we need this extra `agentPoolTyped` var or can we just do:

`agentPoolName = agentPool.(*asocontainerservicev1preview.ManagedClustersAgentPool).AzureName()`

and ditto below